src/server/game/Battlefield/Battlefield.cpp

```diff
@@ -396,6 +396,13 @@ void Battlefield::AskToLeaveQueue(Player* player)
     m_PlayersInQueue[player->GetTeamId()].erase(player->GetGUID());
 }
 
+// Called in WorldSession::HandleHearthAndResurrect
+void Battlefield::PlayerAskToLeave(Player* player)
+{
+    // Player leaving Wintergrasp, teleport to Dalaran.
```

Comment:
The comment should also be updated:
```suggestion
    // Player leaving Wintergrasp, trigger Hearthstone spell
```